### PR TITLE
[7.0] [ADD] stock_picking_partial_invoicing

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
This module was written to extend the invoicing capabilities of Odoo, and allows users to create customer or supplier invoices for part of a delivered Order or received Incoming Shipment.

I am submitting this PR for 7.0, and now starting to prepare it for 8.0.

Regards,
Jordi.

### CODE REVIEW

File: stock_picking_partial_invoicing/AUTHORS.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+Jordi Ballester Alomar <jordi.ballester@eficent.com>
```
Comment (pedrobaeza) [inline stock_picking_partial_invoicing/AUTHORS.txt]: Don't put this file. Put this inside the module description.

File: stock_picking_partial_invoicing/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+To contribute to this module, please visit http://odoo-community.org.
+""",
+    'author': "Eficent,Odoo Community Association (OCA)",
+    'website': 'http://www.eficent.com',
+    'license': 'AGPL-3',
+    "depends": ['stock'],
+    "data": [
+        'wizard/stock_invoice_onshipping.xml',
+        'view/stock.xml',
+
```
Comment (pedrobaeza) [inline stock_picking_partial_invoicing/__openerp__.py:100]: Remove these extra lines.

File: stock_picking_partial_invoicing/model/stock.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import fields, orm
+
+
+class stock_move(orm.Model):
```
Comment (pedrobaeza) [inline stock_picking_partial_invoicing/model/stock.py]: Use CamelCase for class name.

File: stock_picking_partial_invoicing/model/stock.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                                        string='Invoiced quantity',
+                                        type='float'),
+        'invoice_lines': fields.many2many('account.invoice.line',
+                                          'stock_move_invoice_rel',
+                                          'move_id', 'invoice_line_id',
+                                          'Invoice Lines', readonly=True),
+    }
+
+
+class stock_picking(orm.Model):
```
Comment (pedrobaeza) [inline stock_picking_partial_invoicing/model/stock.py]: Use CamelCase for class name.

File: stock_picking_partial_invoicing/wizard/stock_invoice_onshipping.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import fields, orm
+from openerp.tools.translate import _
+import openerp.addons.decimal_precision as dp
+
+
+class stock_invoice_onshipping(orm.TransientModel):
```
Comment (pedrobaeza) [inline stock_picking_partial_invoicing/wizard/stock_invoice_onshipping.py]: Use CamelCase for class name.

File: stock_picking_partial_invoicing/wizard/stock_invoice_onshipping.py [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+            for move in picking.move_lines:
+                product_qty = move.product_uos_qty or move.product_qty
+                if move.invoiced_qty != product_qty:
+                    picking_obj.write(cr, uid, [picking.id],
+                                      {'invoice_state': '2binvoiced'},
+                                      context=context)
+        return res
+
+
+class stock_invoice_onshipping_line(orm.TransientModel):
```
Comment (pedrobaeza) [inline stock_picking_partial_invoicing/wizard/stock_invoice_onshipping.py]: Use CamelCase for class name.

File: stock_picking_partial_invoicing/wizard/stock_invoice_onshipping.py [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+                'Product Unit of Measure'), readonly=True,
+            help='Remaining quantity to invoice'),
+        'price_unit': fields.related(
+            'move_id', 'price_unit', type='float',
+            string='Unit Price', readonly=True),
+        'invoiced_qty': fields.float(
+            'Quantity to invoice', digits_compute=dp.get_precision(
+                'Product Unit of Measure')),
+        'wizard_id': fields.many2one('stock.invoice.onshipping', 'Wizard'),
+    }
```
Comment (pedrobaeza) [inline stock_picking_partial_invoicing/wizard/stock_invoice_onshipping.py]: Put an extra line at the end (PEP8 error).

**Comment by Cedric-Pigeon** [comment]
@jbeficent Is this still PR still useful ?

**Comment by pedrobaeza** [comment]
Closing as very old. If still interested, feel free to reopen.

**Pull request closed without merging**